modepy/spaces.py

```diff
@@ -126,28 +211,41 @@ class QN(FunctionSpace):
     .. automethod:: __init__
     """
 
+    # NOTE: https://github.com/python/mypy/issues/1020
+    def __new__(cls, spatial_dim: int, order: int) -> Any:
+        if spatial_dim == 1:
+            return PN(spatial_dim, order)
+        else:
+            return FunctionSpace.__new__(cls)
+
     def __init__(self, spatial_dim: int, order: int) -> None:
-        super().__init__()
-        self.spatial_dim = spatial_dim
-        self.order = order
+        super().__init__((PN(1, order),) * spatial_dim)
 
     @property
-    def space_dim(self) -> int:
-        return (self.order + 1)**self.spatial_dim
+    def order(self):
+        return self.bases[0].order
 
     def __repr__(self) -> str:
         return (f"{type(self).__name__}("
-            f"spatial_dim={self.spatial_dim}, order={self.order}"
-            ")")
+                f"spatial_dim={self.spatial_dim}, order={self.order}"
+                ")")
 
 
 @space_for_shape.register(Hypercube)
-def _space_for_hypercube(shape: Hypercube, order: int):
-    return QN(shape.dim, order)
-
-# }}}
+def _space_for_hypercube(
```

Comment:
Removed in 9a9ab21.

I kept it around because it would create `QN` spaces. Not sure if anyone in `meshmode` or `grudge` checks for that with `isinstance` or something.

Reply:
Yep, it broke something! Needs inducer/meshmode#290.

Should I add it back for backwards compatibility?

Reply:
Nah, if nothing is broken, it should stay gone.